pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -540,8 +598,19 @@ func (p *provider) Create(ctx context.Context, machine *clusterv1alpha1.Machine,
 
 	userDataSecretName := fmt.Sprintf("userdata-%s-%s", machine.Name, strconv.Itoa(int(time.Now().Unix())))
 
+	var machineDeploymentName string
+	if mdName, err := machineDeploymentNameAndRevisionForMachineGetter(ctx, machine, data.Client)(); err == nil {
+		machineDeploymentName = mdName
+	}
```

Comment:
I am not sure if I follow this logic. 
Why it cannot be:
```
mdName, err := machineDeploymentNameAndRevisionForMachineGetter(ctx, machine, data.Client)()
if err != nil {
    return nil, err
}
```

Reply:
Before this change, we used to create VMs even if getting the MD-name returned error.
So continue with earlier behavior.

Reply:
so if MD-name returns an error then we have a problem. Based on that you create a data volume.